### PR TITLE
[Sprint Maintain] Fix Logic List Domain In Shop Management

### PR DESCRIPTION
**Pull request by Tuanlvt**
[Sprint Maintain] Fix Logic List Domain In Shop Management
Link redmine : https://dev.framgia.com/redmine/issues/123978
Task : Maintain
Effect: N/A

### CODE REVIEW

File: app/src/main/java/com/framgia/forder/screen/shopinfo/ShopinfoContract.java [hunk 1]
```diff
@@ -18,6 +20,18 @@
...
         void onGetListManagerOfShopError(BaseException exception);
+
+        void onApplyToDomainSuccess();
+
+        void onApplyToDomainError(BaseException exception);
+
+        void onLeaveToDomainSuccess();
+
+        void onLeaveToDomainError(BaseException exception);
+
```
Comment (HyperionsThs) [inline app/src/main/java/com/framgia/forder/screen/shopinfo/ShopinfoContract.java]: 2 hàm error đều show toast thì bỏ chung 1 hàm thôi

File: app/src/main/java/com/framgia/forder/screen/shopinfo/ShopinfoPresenter.java [hunk 2]
```diff
@@ -52,7 +66,75 @@ public void call(List<User> users) {
...
+                    @Override
+                    public void call() {
+                        mViewModel.onHideProgressBar();
+                    }
                 });
         mCompositeSubscription.add(subscription);
     }
+
+    @Override
+    public void onApplyToDomain(ApplyShopToDomainRequest applyShopToDomainRequest) {
```
Comment (nguyenhuy95dn) [inline app/src/main/java/com/framgia/forder/screen/shopinfo/ShopinfoPresenter.java]: init User from repo at here

File: app/src/main/java/com/framgia/forder/screen/shopinfo/listdomain/ListDomainAdapter.java [hunk 3]
```diff
@@ -20,12 +23,17 @@
...
+    private final Context mContext;
+    private final List<Domain> mDomains;
+    private final List<ShopInfo> mShopInfos;
+    private final AnimationManager mAnimationManager;
+    private ShopDomainManagementListener mDomainManagementListener;
 
     public ListDomainAdapter(@NonNull Context context) {
         super(context);
-        mShopManagements = new ArrayList<>();
+        this.mContext = context;
```
Comment (nguyenhuy95dn) [inline app/src/main/java/com/framgia/forder/screen/shopinfo/listdomain/ListDomainAdapter.java]: remove this ``this.mContext``

File: app/src/main/java/com/framgia/forder/screen/shopinfo/listdomain/ListDomainAdapter.java [hunk 4]
```diff
@@ -34,41 +42,59 @@ public ListDomainAdapter(@NonNull Context context) {
...
             super(binding.getRoot());
+            this.mContext = context;
+            mDomainManagementListener = shopDomainManagementListener;
             mBinding = binding;
         }
 
-        void bind(ShopManagement shopManagement) {
-            mBinding.setViewModel(new ItemListDomainViewModel(shopManagement));
+        void bind(ShopInfo shopInfo, Domain domain) {
+            mBinding.setViewModel(new ItemListDomainViewModel(mContext, shopInfo, domain,
```
Comment (nguyenhuy95dn) [inline app/src/main/java/com/framgia/forder/screen/shopinfo/listdomain/ListDomainAdapter.java]: as above

File: app/src/main/java/com/framgia/forder/screen/shopinfo/listdomain/ListDomainAdapter.java [hunk 5]
```diff
@@ -34,41 +42,59 @@ public ListDomainAdapter(@NonNull Context context) {
...
-        mShopManagements.addAll(shopManagements);
+        mDomains.clear();
+        mDomains.addAll(domains);
+        mShopInfos.addAll(shopInfos);
         notifyDataSetChanged();
     }
 
     static class ItemViewHolder extends RecyclerView.ViewHolder {
 
+        private final Context mContext;
```
Comment (nguyenhuy95dn) [inline app/src/main/java/com/framgia/forder/screen/shopinfo/listdomain/ListDomainAdapter.java]: remove if not used

File: app/src/main/java/com/framgia/forder/screen/shopinfo/listdomain/ListDomainAdapter.java [hunk 6]
```diff
@@ -34,41 +42,59 @@ public ListDomainAdapter(@NonNull Context context) {
...
         private final ItemListDomainBinding mBinding;
+        private final ShopDomainManagementListener mDomainManagementListener;
 
-        ItemViewHolder(ItemListDomainBinding binding) {
+        ItemViewHolder(Context context, ItemListDomainBinding binding,
+                ShopDomainManagementListener shopDomainManagementListener) {
             super(binding.getRoot());
+            this.mContext = context;
```
Comment (nguyenhuy95dn) [inline app/src/main/java/com/framgia/forder/screen/shopinfo/listdomain/ListDomainAdapter.java]: as above

File: app/src/main/java/com/framgia/forder/screen/shopinfo/listdomain/ListDomainAdapter.java (hunk 3 above)
Comment (Tuanlvt) [inline app/src/main/java/com/framgia/forder/screen/shopinfo/listdomain/ListDomainAdapter.java]: Use it !

File: app/src/main/java/com/framgia/forder/screen/shopinfo/ShopinfoPresenter.java [hunk 7]
```diff
@@ -52,7 +62,74 @@ public void call(List<User> users) {
...
+                            }
+                        }, new SafetyError() {
+                            @Override
+                            public void onSafetyError(BaseException error) {
+                                mViewModel.onApplyOrLeaveToDomainError(error);
+                            }
+                        }, new Action0() {
+                            @Override
+                            public void call() {
+                                mViewModel.onHideProgressBar();
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/shopinfo/ShopinfoPresenter.java]: pls call this line on ``doAfterSubscribe``

File: app/src/main/java/com/framgia/forder/screen/shopinfo/ShopinfoPresenter.java [hunk 8]
```diff
@@ -52,7 +62,74 @@ public void call(List<User> users) {
...
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/shopinfo/ShopinfoPresenter.java]: like above

**Pull request merged**